File: WebApplication/src/app/leaderboard/irp-search/SearchIrpsRequestDto.ts
import { removeUndefinedKeyValues } from "../../_common/jsonHelpers"

export enum SearchOnField {
  Bib,
  FirstName,
  LastName,
  FullName
}

export class SearchIrpsRequestDto {
  constructor(
    public searchOn: SearchOnField,
    public searchTerm: string,
    public raceId: number | null,
    public courseId: number | null
  ) {}

  public getAsParamsObject = () => {

    const params = {
      searchOn: this.searchOn,
      searchTerm: this.searchTerm,
      raceId: this.raceId,
      courseId: this.courseId,
    }

    return removeUndefinedKeyValues(params)
  }
}

File: Api/Orchestration/SearchIrps/SearchIrpsRequestDto.cs
namespace Api.Orchestration.SearchIrps;

public record SearchIrpsRequestDto
{
    public int? RaceId { get; init; }

    public int? CourseId { get; init; }

    public SearchOnField SearchOn { get; init; }

    public string SearchTerm { get; init; }
}
